ingenialink/ethercat/network.py

```diff
@@ -95,18 +102,36 @@ def __init__(
         self._ecat_master: pysoem.CdefMaster = pysoem.Master()
         self._ecat_master.sdo_read_timeout = int(1_000_000 * connection_timeout)
         self._ecat_master.sdo_write_timeout = int(1_000_000 * connection_timeout)
+        self._ecat_master.manual_state_change = self.MANUAL_STATE_CHANGE
         self.__is_master_running = False
+        self.__last_init_nodes: List[int] = []
 
     def scan_slaves(self) -> List[int]:
-        """Scans for nodes in the network.
+        """Scans for nodes in the network. If any node is already connected scan can not be done.
 
         Returns:
             List containing all the detected node IDs.
 
+        Raises:
+            ILError: If any node is already connected.
+
+        """
+        if self.servos:
+            raise ILError("Some nodes are already connected")
+        if not self.__is_master_running:
+            self._start_master()
+        self.__init_nodes()
+        return self.__last_init_nodes
+
+    def __init_nodes(self) -> None:
+        """Init all the nodes and set already connected nodes to PreOp state.
+        Also fill `__last_init_nodes` attribute.
         """
-        self._start_master()
+        self._ecat_master.slaves = []  # That is because of a bug of pysoem
```

Comment:
Has it been reported? Can we link it in the code?

Reply:
I just added an issue to pysoem: https://github.com/bnjmnp/pysoem/issues/128